src/Database.cpp

```diff
@@ -0,0 +1,460 @@
+#include "Database.hpp"
+#include "contrib/sqlite/sqlite3pp.h"
+#include "contrib/bcrypt/BCrypt.hpp">
+#include "contrib/json11.hpp"
+#include <fstream>
+#include <iostream>
+#include "CNProtocol.hpp"
+#include <string>
+
+#include <regex>
+#include "CNStructs.hpp"
+#include "settings.hpp"
+#include "Player.hpp"
+#include <list>
+#include "CNStructs.hpp"
+
+using namespace json11;
+
+//TODO: replace this sqlite wrapper with something better, clean up queries, get rid of json
+
+sqlite3pp::database db;
+
+void Database::open() {
+	//checking if database file exists
+	std::ifstream file;
+	file.open("data.db");
+	if (file) {
+		file.close();
+		//if exists, assign it		
+		db = sqlite3pp::database ("data.db");
+
+		DEBUGLOG(
+			std::cout << "Database in operation" << std::endl;
+		)
+	}
+	else {
+		//if doesn't, create all the tables
+		DEBUGLOG(
+			std::cout << "Creating new database" << std::endl;
+		)
+		db = sqlite3pp::database("data.db");
+		db.execute("CREATE TABLE Accounts(AccountID INTEGER PRIMARY KEY AUTOINCREMENT, Login TEXT NOT NULL, Password TEXT NOT NULL);");
+		db.execute("CREATE TABLE Players(PlayerID INTEGER PRIMARY KEY AUTOINCREMENT, AccountID INTEGER NOT NULL, Slot INTEGER NOT NULL, FirstName TEXT NOT NULL, LastName TEXT NOT NULL, CharData TEXT NOT NULL);");
+		DEBUGLOG(
+			std::cout << "Done" << std::endl;
+		)
+	}
+	
+
+}
+
+bool Database::isLoginDataGood(std::string login, std::string password) {
+	std::regex loginRegex("^([A-Za-z\\d_\\-]){5,20}$");
```

Comment:
I reckon the regex check here is to "prevent" SQLi, but we can't guarantee that this code isn't touched by anyone else. If this check is somehow changed or removed, the entire database would be in shambles.